library/index.c

```diff
@@ -323,24 +323,25 @@ int eblob_index_blocks_fill(struct eblob_base_ctl *bctl)
 }
 
 
-static struct eblob_disk_control *eblob_find_on_disk(struct eblob_backend *b,
-		struct eblob_base_ctl *bctl, struct eblob_disk_control *dc,
+static int eblob_find_on_disk(struct eblob_backend *b,
```

Comment:
I made a mistake, bsearch() is used to locate block within range tree (it is sorted array filled at eblob start) if bloom filter didn't reject it first, and then linear search is used to locate given key in the index block (40 entries by default) if it exists.